### ISSUE TITLE
Fire TV support?

### ISSUE THREAD

**Issue by Rudxain** [termux/termux-app#3477]
> [The app could later be optimised to work on the Fire TV and Fire TV Stick devices.](https://github.com/termux/termux-app/issues/326#issuecomment-366207778)

Will there ever be support for those devices?

I installed `termux-app_v0.118.0+github-debug_armeabi-v7a.apk` via `adb install`, on a 3rd-gen. It worked! But the OS virtual-keyboard disappears after a split-second, for some reason, so I can't type anything. The RC navigation-buttons don't work either, so I can't use the Termux keyboard.

The only workarounds are `adb shell input` or `scrcpy` (not tested yet), to send tap and text events

**Comment by twaik** [comment]
As a workaround you can use sshd and some ssh client on other device.

**Comment by agnostic-apollo** [comment]
You could try swiping left on extra keys with mouse to open text input mode, keyboard may open there. And then try using `enforce-char-based-input = true`.

https://wiki.termux.com/wiki/Terminal_Settings#Workaround_issues_with_some_keyboards

**Comment by Rudxain** [comment]
> You could try swiping left on extra keys with mouse to open text input mode, keyboard may open there.

It works! But now it's impossible to close the keyboard.

> And then try using `enforce-char-based-input = true`.

I'll try editing the properties outside the device, because the virtual keyboard is impossible to use with `nano`

I think I should install `ssh` or use a Bluetooth keyboard (which I don't currently have)

**Comment by agnostic-apollo** [comment]
Isn't there a back button shown in the UI?

You could also use keyboard button in left drawer or `KEYBOARD` extra key to toggle. But bluetooth keyboard would be better for everything, even non-termux. There are usb wireless keyboards available too, probably cheaper.

**Comment by Rudxain** [comment]
> Isn't there a back button shown in the UI?

No, only physical RC. Both RC back-button and `adb shell input keyevent 4` are interpreted as `ENTER` (line break) by Bash and Nano. Which makes even less sense, because there's a dedicated "Next/Send" button that does the same thing.
 
> You could also use keyboard button in left drawer

What do you mean? 🤔

> or `KEYBOARD` extra key to toggle.

The problem is that the virtual keyboard **blocks all touch input**, even when using ADB, so even if there was an extra-key on Termux I couldn't press it.

My 1st thought was to swipe right, to close char-input and go back to extra-keys, but the swipe doesn't work either.

I can exit Bash by sending `exit`, but I **can't escape `nano`** with a literal (there's no Ctrl key) `^X`. This reminds me of the "can't exit Vim" problem, haha.

> bluetooth keyboard would be better for everything

I agree! It would also make typing on phones easier.

> usb wireless keyboards 

Unfortunately, the AFTV device I'm using only has 1 microUSB port, and it's the power supply. I'm not sure if a USB-hub would work in this case

**Comment by agnostic-apollo** [comment]
>What do you mean? 

Swipe right from left edge of screen.

>I'm using only has 1 microUSB port, and it's the power supply. I'm not sure if a USB-hub would work in this case

You probably need a usb otg dongle that accepts power input.

**Comment by Rudxain** [comment]
> Swipe right from left edge of screen.

Ah, yes. I use that to open new sessions. The keyboard toggle is useful sometimes.
 
> You probably need a usb otg dongle that accepts power input.

That's a good idea, but I'll probably use BT keyboard.

Thanks for the help anyways! 👍 :)

**Comment by agnostic-apollo** [comment]
You are welcome. You can try running `echo "enforce-char-based-input=true" >> .termux/termux.properties` and then `termux-reload-settings` if ctrl doesn't work.

**Comment by Rudxain** [comment]
I'll try that. Thanks.

While looking at my `termux.properties`, I found `# soft-keyboard-toggle-behaviour = enable/disable`, which I thought would be useful in this situation, but it would only work if the keyboard didn't block touches

**Comment by Rudxain** [comment]
`enforce-char-based-input=true` now makes the keyboard auto-open without swiping extra-keys (nice!), but the keyboard still can't be closed :(

**Comment by agnostic-apollo** [comment]
You can try adding `KEYBOARD` key to extra keys to toggle.

**Comment by Rudxain** [comment]
As I said previously, the keyboard blocks touch input.

However, I somehow managed to "glitch" the keyboard to close itself, but I only did it once, and it only seems to work while `enforce-char-based-input=false`. I still have no idea what I did or how the bug works.

A similar glitch allowed me to irreversibly bypass the text-box (the one that shows after sliding extra-keys), and allowed me to type directly on the terminal (default typing mode), as if the text-box didn't exist.

While the keyboard is closed, the up-&-down RC arrow-keys somehow work.

Backspace only works on text-box, so I can't undo typings while `enforce-char-based-input=true`

**Comment by vishal369m** [comment]
I used it and got a way to do it without the keyboard appearing and no extra keyboard needed

**Comment by vishal369m** [comment]
Just go the the termux setting and turn off the virtual keyboard
Then download firetv remote from Google play store and connect to your fire tv 
Then use the keyboard button on the top right corner of the app on your mobile phone and type anything you want 
To press enter you have to click the circular ok button in the remote or in the app

**Comment by Rudxain** [comment]
I'm now [questioning if amazon actually deserves support](https://consumerrights.wiki/w/Amazon)
